### ISSUE TITLE
`watchOnTermination` setting not respected

### ISSUE THREAD

**Issue by mrdziuban** [sbt/sbt#6861]
## steps

Using SBT 1.6.2, add the following `watchOnTermination` setting to a `build.sbt` file:

```scala
ThisBuild / watchOnTermination := { (action, cmd, times, state) =>
  println(s"watchOnTermination: action: $action, cmd: $cmd, times: $times, state: $state")
  state
}
```

Then load SBT in the project's directory, run `~compile`, and then hit enter to stop triggered execution.

## problem

My `watchOnTermination` function is not called after I hit enter to stop triggered execution

## expectation

The `watchOnTermination` function should be called when I hit enter and I should see my `println` message in the console

## notes

Apologies if I'm misunderstanding how this setting is supposed to work. I see that it's [not documented with the other `watch*` settings](https://www.scala-sbt.org/1.x/docs/Triggered-Execution.html#Configuration), I just learned about it by reading through the source code.

**Comment by eed3si9n** [comment]
I am not super familiar with the watch implementation, but maybe the fact that it was omitted from the documentation was an indication that it's no longer supported.

@eatkins might have a better context of this if he still remembers.

**Comment by eatkins** [comment]
Should be fixed by #6870

**Comment by mrdziuban** [comment]
Thank you for the quick fix!
